src/utils/errors.ts

```diff
@@ -1,19 +1,19 @@
 import { parseJSONRPCError } from './parsing'
 
-// some clients use -32000 for both server/user errors, so message match is needed
-const EVM_USER_ERROR_MSGS = ['execution reverted', 'stack underflow', 'cannot be found', 'missing trie node']
+// Messages thrown by evm clients like Geth that are node fault, not user fault.
+const EVM_SERVER_ERROR_MSGS = ['connection error']
 
 // EVM clients rely on JsonRpc standard, so we check for those errors.
 export function isUserErrorEVM(response: string): boolean {
   try {
     const { code, message } = parseJSONRPCError(response)
 
-    const userError = EVM_USER_ERROR_MSGS.some((err) => message.includes(err))
+    const serverError = EVM_SERVER_ERROR_MSGS.some((err) => message.includes(err))
 
     // 3 is execution error
     // -32000 itself can be thrown for what are server errors
     // all other -32000 are user errors
-    return userError || code === 3 || code < -32000
+    return (!serverError && code === -32000) || code === 3 || code < -32000
```

Comment:
anything thats not a server error & has error -32k is assumed to be an user error, not node error.